main.tf

```diff
@@ -2,41 +2,43 @@ resource "aws_vpc" "main" {
   cidr_block = "10.0.0.0/16"
 
 }
-resource "aws_subnet" "public_subnet1" {
+resource "aws_subnet" "public_subnet" {
+  count             = 2
   vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.1.0/24"
-  availability_zone = "us-east-1a"
+  cidr_block        = var.public_cidr[count.index]
+  availability_zone = var.availability_zones[count.index]
 
   tags = {
-    name = "public_subnet1"
+    name = "public_subnet${count.index}"
   }
 
 }
-resource "aws_subnet" "public_subnet2" {
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.2.0/24"
-  availability_zone = "us-east-1b"
-  tags = {
-    name = "public_subnet2"
-  }
-}
+#resource "aws_subnet" "public_subnet2" {
+#  vpc_id            = aws_vpc.main.id
+#  cidr_block        = "10.0.2.0/24"
+#  availability_zone = "us-east-1b"
+#  tags = {
+#    name = "public_subnet2"
+#  }
+#}
 
-resource "aws_subnet" "private_subnet1" {
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.3.0/24"
-  availability_zone = "us-east-1a"
-  tags = {
-    name = "private_subnet1"
-  }
-}
-resource "aws_subnet" "private_subnet2" {
+resource "aws_subnet" "private_subnet" {
```

Comment:
could you please change count = 2 to count = length(var.private_cidr) and also change for resources that has count  and is related to the private subnet.